docs/CommandGuide/handshake-runner.md

```diff
@@ -0,0 +1,67 @@
+**handshake-runner** — Run restricted std MLIR and CIRCT handshake IR
+
+## SYNOPSIS
+
+| **handshake-runner** \[_options_] \[_filename_] \[_arguments_]
+
+## DESCRIPTION
+
+This application executes a function in the given MLIR module. Arguments to the 
+function are passed on the command line and results are returned on stdout. 
+Memref types are specified as a comma-separated list of values. This particular 
+tool is use to check the validity of Standard-to-Handshake conversion in CIRCT.
+
+### Example
+
+The following MLIR module first convert to Handshake IR with the circt-opt tool 
+as `circt-opt -create-dataflow <file-name>`. And then the converted handshake IR
+can be run by the handshake-runner with a command line argument "2" as 
+`handshake-runner <file-name> 2` and produces the expected result "1".
+```
+module {
+  func @main(%arg0: index) -> (i8) {
+    %0 = alloc() : memref<10xi8>
+    %c1 = constant 1 : i8
+         store %c1, %0[%arg0] : memref<10xi8>
+         %1 = load %0[%arg0] : memref<10xi8>
+         return %1 : i8
+  }
+}
```

Comment:
The indentation is a little weird on my side.

Reply:
It is picked from a testcase see [handshake-runner/loadstore.mlir](https://github.com/llvm/circt/blob/main/test/handshake-runner/loadstore.mlir.).